### PR TITLE
Add non-aggregate metric types and support metric references

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #58 

Preview:
Screenshot 2021-06-30 at 20 08 11

Preview YAML:
```
        meta:
          metrics:
            my_total_customers:
              type: count_distinct
            my_total_customers_plus:
              type: number
              sql: 2 + ${my_total_customers}
```

### CODE REVIEW

File: docs/docs/references/metrics.md [hunk 1]
```diff
@@ -65,3 +69,22 @@ metrics:
     type: count_distinct
     sql: "IF(${subscriptions.is_active}, ${user_id}, NULL)"
 ```
+
+## Using custom SQL in non-aggregate metrics
```
Comment (owlas) [inline docs/docs/references/metrics.md:73]: I wonder if non-aggregate is the right name for these derived metrics? It's more like you can just write whatever you want as long as it results in an aggregated metric (so it kinda is aggregated?).

Maybe "custom aggregations" "derived aggregations"

or auto-aggregate vs. manual-aggregate

Comment (owlas) [reply docs/docs/references/metrics.md:73]: I get non-aggregate means it doesn't aggregate **for you** but this metric (like all metrics) actually will be an aggregate value. So a bit confusing

Comment (ZeRego) [reply docs/docs/references/metrics.md:73]: Not sure if I agree with 'manual-aggregate'. Can't we have a metric that isn't an aggregate ? For example for the boolean type, checking if a combination of values are present.  
In that case we might need something like `aggregate` vs `freestyle`

Comment (owlas) [reply docs/docs/references/metrics.md:73]: I love freestyle haha

Comment (owlas) [reply docs/docs/references/metrics.md:73]: Ok, let's stick with Looker names. Since they are well known

**Review by owlas — APPROVED** [review]

**Comment by owlas** [comment]
Closes #59

**Pull request merged**